File: bindings/swift/Sources/RockboxFFICore/Enums.swift
// Note the two *different* ReplayGain encodings in the C ABI.

/// Values for `Dsp.setReplaygain` (native Rockbox).
public enum DspReplayGainMode: Int32 {
    case track = 0
    case album = 1
    case shuffle = 2
    case off = 3
}

/// Values for `Player.setReplaygain`.
public enum ReplayGainMode: Int32 {
    case off = 0
    case track = 1
    case album = 2
}

public enum CrossfadeMode: Int32 {
    case off = 0
    case autoSkip = 1
    case manualSkip = 2
    case shuffle = 3
    case shuffleOrManual = 4
    case always = 5
}

public enum MixMode: Int32 {
    case crossfade = 0
    case mix = 1
}

/// Where to place inserted / imported tracks in the queue.
public enum InsertPosition: Int32 {
    case prepend = 0
    case insert = 1
    case insertNext = 2
    case insertLast = 3
    case insertShuffled = 4
    case insertLastShuffled = 5
    case replace = 6
    /// Explicit index — uses the `index` argument.
    case index = 7
}

public enum ChannelConfig: Int32 {
    case stereo = 0
    case mono = 1
    case custom = 2
    case monoLeft = 3
    case monoRight = 4
    case karaoke = 5
    case swap = 6
}

/// Built-in EQ presets for `Player.setEqPreset`.
public enum EqPreset: Int32 {
    case flat = 0
    case acoustic = 1
    case bassBoost = 2
    case bassReducer = 3
    case classical = 4
    case dance = 5
    case deep = 6
    case electronic = 7
    case hipHop = 8
    case jazz = 9
    case latin = 10
    case loudness = 11
    case lounge = 12
    case piano = 13
    case pop = 14
    case rnb = 15
    case rock = 16
    case smallSpeakers = 17
    case trebleBoost = 18
    case trebleReducer = 19
    case vocalBoost = 20
}

/// Channel mixing modes for `Player.setChannelMode`.
public enum ChannelMode: Int32 {
    case stereo = 0
    case mono = 1
    case custom = 2
    case monoLeft = 3
    case monoRight = 4
    case karaoke = 5
    case swap = 6
}

/// ABI major version of the loaded library (bumped on breaking changes).
public func abiVersion() -> UInt32 {
    Lib.shared.abiVersion()
}

File: bindings/python/src/rockbox_ffi/enums.py
"""Enum constants for the Rockbox FFI.

Note the two *different* ReplayGain encodings in the C ABI:

* The **DSP-level** call ``rb_dsp_set_replaygain`` (exposed via
  :meth:`rockbox_ffi.dsp.Dsp.set_replaygain`) uses the Rockbox-native values:
  ``TRACK=0, ALBUM=1, SHUFFLE=2, OFF=3`` — see :class:`DspReplayGainMode`.
* The **player-level** call ``rb_player_set_replaygain`` uses a simpler
  encoding: ``OFF=0, TRACK=1, ALBUM=2`` — see :class:`ReplayGainMode`.
"""

from enum import IntEnum


class DspReplayGainMode(IntEnum):
    """Values for :meth:`rockbox_ffi.dsp.Dsp.set_replaygain` (native Rockbox)."""

    TRACK = 0
    ALBUM = 1
    SHUFFLE = 2
    OFF = 3


class ReplayGainMode(IntEnum):
    """Values for :meth:`rockbox_ffi.player.Player.set_replaygain`."""

    OFF = 0
    TRACK = 1
    ALBUM = 2


class RepeatMode(IntEnum):
    """Values for :meth:`rockbox_ffi.player.Player.set_repeat`."""

    OFF = 0
    ONE = 1
    ALL = 2


class CrossfadeMode(IntEnum):
    OFF = 0
    AUTO_SKIP = 1
    MANUAL_SKIP = 2
    SHUFFLE = 3
    SHUFFLE_OR_MANUAL = 4
    ALWAYS = 5


class MixMode(IntEnum):
    CROSSFADE = 0
    MIX = 1


class InsertPosition(IntEnum):
    """Values for :meth:`rockbox_ffi.player.Player.insert` and
    :meth:`~rockbox_ffi.player.Player.import_m3u`.

    ``INDEX`` uses the explicit ``index`` argument; all others ignore it.
    """

    PREPEND = 0
    INSERT = 1
    INSERT_NEXT = 2
    INSERT_LAST = 3
    INSERT_SHUFFLED = 4
    INSERT_LAST_SHUFFLED = 5
    REPLACE = 6
    INDEX = 7


class ChannelConfig(IntEnum):
    STEREO = 0
    MONO = 1
    CUSTOM = 2
    MONO_LEFT = 3
    MONO_RIGHT = 4
    KARAOKE = 5
    SWAP = 6


class EqPreset(IntEnum):
    """Values for :meth:`rockbox_ffi.player.Player.set_eq_preset`."""

    FLAT = 0
    ACOUSTIC = 1
    BASS_BOOST = 2
    BASS_REDUCER = 3
    CLASSICAL = 4
    DANCE = 5
    DEEP = 6
    ELECTRONIC = 7
    HIP_HOP = 8
    JAZZ = 9
    LATIN = 10
    LOUDNESS = 11
    LOUNGE = 12
    PIANO = 13
    POP = 14
    RNB = 15
    ROCK = 16
    SMALL_SPEAKERS = 17
    TREBLE_BOOST = 18
    TREBLE_REDUCER = 19
    VOCAL_BOOST = 20


class ChannelMode(IntEnum):
    """Values for :meth:`rockbox_ffi.player.Player.set_channel_mode`."""

    STEREO = 0
    MONO = 1
    CUSTOM = 2
    MONO_LEFT = 3
    MONO_RIGHT = 4
    KARAOKE = 5
    SWAP = 6


class CrossfeedMode(IntEnum):
    """Values for :meth:`rockbox_ffi.player.Player.set_crossfeed`."""

    OFF = 0
    MEIER = 1
    CUSTOM = 2
